antidox/wikiwatcher.py

```diff
@@ -4,21 +4,63 @@
 from __future__ import division
 from __future__ import print_function
 
-import argparse
 import json
-
+import pprint
+import argparse
+import requests
 import sseclient
+from googleapiclient import errors as google_api_errors
+import clean
+import perspective
 
 
+# pylint: disable=fixme, too-many-locals
 def log_event(change):
   """Logs event by printing.
 
   Args:
     change: a json object with the wikimedia change record.
   """
-  print(
-      u'user:{user} namespace:{namespace} bot:{bot} comment:{comment} title:{title}'
-      .format(**change))
+  # print(
+  #     u'user:{user} namespace:{namespace} bot:{bot} comment:{comment} title:{title}'
+  #     .format(**change))
+  print('\n########## change:')
+  from_id = (str(change['revision']['old']))
+  to_id = (str(change['revision']['new']))
+  page = ("https://en.wikipedia.org/w/api.php?action=compare&fromrev="
+          + from_id + "&torev=" + to_id + "&format=json")
+  get_page = requests.get(page)
+  response = json.loads(get_page.content)
+  revision = response['compare']['*']
+
+  text = clean.content_clean(revision).split("\n")
```

Comment:
This splits `text` by newline. I think for doxing detection, it makes more sense to analyze the entire change at a time, so I don't think we need to do the `split('\n')` or do the for loop over all the lines.

Reply:
without it, it's going to print letter by letter.

Reply:
Did you also remove the `for` loop? If you still have the for loop, it will loop over all the characters in the string.

I was saying you should do a single Perspective and DLP request for the entire change, so you don't need the for loop or the `split()` at all.